### ISSUE TITLE
CupertinoNumericTransition Widget

### ISSUE THREAD

**Issue by naamapps** [flutter/flutter#156906]
### Use case

Since iOS 17, SwiftUI has a built-in way to animate numbers.

This is what it looks like.

![animating-number-changes-in-swiftui-new](https://github.com/user-attachments/assets/8bea0f09-062c-4f00-9f50-c17bc441e623)

There is no built-in way to achieve this in flutter. 

### Proposal

Having this transition built into Flutter would be huge for a good UX/UI boost while also easing the switch to Flutter for SwiftUI developers.
Having this as a built in widget will also provide developers a way for their apps to fit even more inside the Apple ecosystem - ensuring apps have the correct fidelity.

**Comment by Piinks** [comment]
Might be of interest to @johnpryan @hannah-hyj

**Comment by MitchellGoodwin** [comment]
My gut feeling is that this may be a low priority for Cupertino and would be better served by a package. Some questions I have:

Are there any examples of this being used in any large enterprise apps? Apple or otherwise? If it's only being used in some places and is not strictly necessary to get an iOS native looking app it may not be a priority for us to make. We'd still review a PR if somebody took the time to do so. 

If somebody was to make this widget themselves would they have significant difficulties? Outside of this widget, if implementing it is a larger challenge than expected with our animation APIs, that may be something we need to look into.

**Comment by MaherSafadii** [comment]
> My gut feeling is that this may be a low priority for Cupertino and would be better served by a package. Some questions I have:
> 
> Are there any examples of this being used in any large enterprise apps? Apple or otherwise? If it's only being used in some places and is not strictly necessary to get an iOS native looking app it may not be a priority for us to make. We'd still review a PR if somebody took the time to do so.
> 
> If somebody was to make this widget themselves would they have significant difficulties? Outside of this widget, if implementing it is a larger challenge than expected with our animation APIs, that may be something we need to look into.

I agree, I don't see it used in large enterprise apps, neither is it a high priority iOS UX feature that users are expecting.
